antiplag/views.py

```diff
@@ -22,13 +24,24 @@ def post(self, request):
 
         # create new submission
         submission = Submission.objects.create(
-            status=Submission.SubmissionStatus.PENDING
+            status=Submission.SubmissionStatus.PENDING,
+            email = request.POST.get("email", None)
         )
 
         # check for request content type
         is_file = CONTENT_TYPE_FILE in request.content_type
         is_text = CONTENT_TYPE_TEXT in request.content_type
 
+        # check if email is valid
+        if submission.email is not None:
```

Comment:
What about moving this to the top of this function? If the request contains invalid email, submission would be long saved in the DB.